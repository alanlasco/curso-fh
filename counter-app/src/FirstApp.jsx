import PropTypes from "prop-types";
import React from "react"; // Agrega esta lÃ­nea
const say = () => {
  return "Hola ALan";
};

export const FirstApp = ({ title, subTitle, name }) => {
  //   const newMessage = {
  //     message: "alan",
  //     title: "lasco",
  //   };

  return (
    <>
      {/*fragmento */}
      {/* <code>{JSON.stringify(newMessage)}</code> */}
      <h1 data-testid="test-id">{title}</h1>
      <p>{subTitle}</p>
      <h2>{say()}</h2>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: "pipo",
  subTitle: "no hay subtitulo",
  title: "no hay titulo",
};
