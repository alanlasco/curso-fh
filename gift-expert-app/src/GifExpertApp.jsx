import { useState } from "react";
import { AddCategory } from "./components/AddCategory";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const addCategory = () => {
    setCategories(["Dragon Ball", ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory SetCategories={setCategories} />
      {/* <button onClick={addCategory}>Agregar</button> */}
      <ol>
        {categories.map((category) => {
          return <li key={category}> {category} </li>;
        })}
      </ol>
    </>
  );
};
