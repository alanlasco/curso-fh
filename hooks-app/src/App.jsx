import { CounterApp } from "./use-State/CounterApp";

function App() {
  return (
    <div className="App">
      <h1>HooksApp</h1>
      <CounterApp />
    </div>
  );
}

export default App;
