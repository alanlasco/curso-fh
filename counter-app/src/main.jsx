import React from "react";
import ReactDOM from "react-dom/client";
import { App } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
import "./index.css";
import { CounterApp } from "./CounterApp";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    <App />
    {/* <FirstApp name="alan" surname="lasco" /> */}
    <FirstApp title="Hola, si anda" />
    {/* <CounterApp value={0} /> */}
  </React.StrictMode>
);
