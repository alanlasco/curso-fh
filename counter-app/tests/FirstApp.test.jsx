import { FirstApp } from "../src/FirstApp";
import React from "react"; // Agrega esta línea
import { render } from "@testing-library/react";

describe("Pruebas en <FirstApp />", () => {
  test("debe hacer match con el snapshot", () => {
    const t = "hola soy goku";
    const { container } = render(<FirstApp title={t} />);
    expect(container).toMatchSnapshot();

    // se asegura que al dia de mañana no vaya a cambiar, evalua todo
  });
  test("debe mostrar el titulo en h1", () => {
    const title = "Hola soy vegeta";

    const { container, getByText } = render(<FirstApp title={title} />);

    expect(getByText(title)).toBeTruthy();
    const h1 = container.querySelector("h1");
    expect(h1.innerHTML).toContain(title);
  });
});
