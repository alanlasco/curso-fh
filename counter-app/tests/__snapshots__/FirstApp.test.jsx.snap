// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    hola soy goku
  </h1>
  <p>
    no hay subtitulo
  </p>
  <h2>
    Hola ALan
  </h2>
</div>
`;
